File: 1207. Unique Number of Occurrences/soln1.py
# Runtime
# 47 ms
# Beats 38.67% of users with Python3

# Memory
# 17.56 MB
# Beats 9.52% of users with Python3

class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        if len(arr) == 1:
            return True
        nums = {}
        for num in arr:
            if num in nums:
                nums[num] = nums.get(num) + 1
            else:
                nums[num] = 1
        count = []
        for num in nums:
            count.append(nums[num])
        count.sort()
        for i in range(1, len(count)):
            if count[i] == count[i-1]:
                return False
        return True
        

File: 451. Sort Characters By Frequency/soln1.cpp
// Runtime 10ms
// Beats 68.64% of users with C++

// Memory 10.88MB
// Beats 24.38% of users with C++

class Solution {
public:
    string frequencySort(string s) {
        
        vector<int> count(75,0);
        map<int,string> m;
    
        for (long int i = 0; i < s.size(); i++){ 
            count[s[i]-'0']++;
        }

        for (int i = 0; i < 75; i++){
            cout << count[i] << " ";
        }
        cout << endl;
        for (int i = 0; i < 75; i++){
            if (count[i] > 0){
                if (m[count[i]] == ""){
                    m[count[i]] = i + '0';
                }
                else{
                    m[count[i]] += i + '0';
                }
            }
            
        }
        sort(count.begin(),count.end());
        string output = "";
        for (int i = 74; i >= 0; i--){

            if (count[i] == 0) break;
            if (i < 74 && count[i] == count[i+1]) continue;

            for (int j = 0; j < m[count[i]].size(); j++){
                for (int k = 0; k < count[i]; k++){
                    output += m[count[i]][j];
                }
            }
            
        }

        return output;
    }
};
